sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12-file-detail.ts

```diff
@@ -1,114 +1,361 @@
-          record.append(disbursementValue.valueCode, 4);
-          record.appendWithStartFiller(
-            round(disbursementValue.valueAmount),
-            8,
-            NUMBER_FILLER,
-          );
-        },
-      );
-      return record.toString();
-    });
-    return records.join(END_OF_LINE);
+    const record = new AdvancedStringBuilder(
+      SPACE_FILLER,
+      NUMBER_FILLER,
+      SPACE_FILLER,
+    );
+    record.appendNumberWithFiller(this.assessmentId, 10);
+    record.appendNumberWithFiller(this.disbursementId, 10);
+    record.append(this.applicationNumber, 10);
+    record.appendOptionalStringWithFiller(this.institutionStudentNumber, 12);
+    record.append(this.sin, 9);
+    record.appendStringWithFiller(this.studentLastName, 25);
+    record.appendOptionalStringWithFiller(this.studentGivenName, 15);
+    record.appendDate(this.studentBirthDate, DATE_FORMAT);
+    record.appendStringWithFiller(this.studentGroupCode, 4);
+    record.appendOptionalStringWithFiller(this.studentMaritalStatusCode, 4);
+    record.appendOptionalStringWithFiller(this.studentDisabilityStatusCode, 4);
+    record.appendOptionalStringWithFiller(
+      this.applicationDisabilityStatusFlag,
+      1,
+    );
+    record.appendStringWithFiller(this.addressInfo.addressLine1, 25);
+    record.appendOptionalStringWithFiller(this.addressInfo.addressLine2, 25);
+    record.appendStringWithFiller(this.addressInfo.city, 25);
+    record.appendStringWithFiller(this.addressInfo.provinceState, 4);
+    record.appendStringWithFiller(this.addressInfo.postalCode, 16);
+    record.appendStringWithFiller(this.programName, 25);
+    record.appendStringWithFiller(this.programDescription, 50);
+    record.appendStringWithFiller(this.credentialType, 25);
+    record.appendNumberWithFiller(this.fieldOfStudyCode, 4);
+    record.appendOptionalStringWithFiller(this.areaOfStudyCode, 4);
+    record.appendOptionalStringWithFiller(this.levelOfStudyCode, 4);
+    record.appendNumberWithFiller(this.currentProgramYear, 2);
+    record.append(this.cipCode, 6);
+    record.appendWithStartFiller(this.nocCode ?? 0, 5, NUMBER_FILLER); // NOC code can only be number in program.
+    record.appendOptionalStringWithFiller(this.sabcCode, 4);
+    record.appendOptionalStringWithFiller(this.institutionProgramCode, 25);
+    record.append(this.programLength, 1);
+    record.appendDate(this.studyStartDate, DATE_FORMAT);
+    record.appendDate(this.studyEndDate, DATE_FORMAT);
+    record.appendNumberWithFiller(this.tuitionFees, 10);
+    record.appendNumberWithFiller(this.booksAndSuppliesCost, 10);
+    record.appendNumberWithFiller(this.mandatoryFees, 10);
+    record.appendNumberWithFiller(this.exceptionExpenses, 10);
+    record.appendNumberWithFiller(this.totalFundedWeeks, 2);
+    record.append(this.courseLoad, 3);
+    record.append(this.offeringIntensityIndicator, 1);
+    record.appendDate(this.applicationSubmittedDate, DATE_FORMAT);
```

Comment:
Just a suggestion, the `DATE_FORMAT` can also be part of the `AdvancedStringBuilder`.

Reply:
I had that thought, but was kinda thinking what if we have to use more than one date format in integration. may be that is like thinking too far ?

Reply:
We can leave the filler as an optional parameter in the method. It is more a suggestion though.
When I saw the "appends", `DATE_FORMAT` was the only one left 😉